File: algorithms/1_easy/F/03731_find_missing_elements/hashset.cpp
class Solution {
public:
    static vector<int> findMissingElements(const vector<int>& nums) {
        const auto set = std::unordered_set<int>(nums.begin(), nums.end());
        const int min = *std::min_element(nums.begin(), nums.end());
        const int max = *std::max_element(nums.begin(), nums.end());
        const int64_t width = static_cast<int64_t>(max) - static_cast<int64_t>(min) + 1;
        const size_t res_len = static_cast<size_t>(width) - set.size();
        auto res = vector<int>();
        res.reserve(res_len);

        for (int val = min; val != max; val += 1) {
            if (!set.contains(val)) {
                res.push_back(val);
            }
        }

        return res;
    }
};

File: algorithms/1_easy/F/03731_find_missing_elements/hashset.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_missing_elements(nums: Vec<i32>) -> Vec<i32> {
        assert!(num.len() > 0);
        let set: HashSet<i32> = nums.iter().map(|x| *x).collect();
        let max: i32 = *nums.iter().max().unwrap();
        let min: i32 = *nums.iter().min().unwrap();
        let width = (max as i64) - (min as i64) + 1;
        let res_len = (width as usize) - set.len();
        let mut res: Vec<i32> = Vec::with_capacity(res_len);

        for val in min..max {
            if !set.contains(&val) {
                res.push(val);
            }
        }

        res
    }
}
